tests/index.ts

```diff
@@ -433,8 +433,8 @@ export function tryDecodeBoolean(description: ParameterDescription, data: BytesL
 	return { result: !!decoded, consumed: 32 }
 }
 
-export function tryDecodeSmallNumber(description: ParameterDescription, data: BytesLike): { result: UInt8 | UInt16 | UInt24 | UInt32 | UInt40 | UInt48 | Int8 | Int16 | Int24 | Int32 | Int40 | Int48, consumed: number } | null {
-	const match = /^(u?)int(\d*)$/.exec(description.type)
+export function tryDecodeSmallNumber(description: ParameterDescription, data: BytesLike): { result: UInt8 | UInt16 | UInt24 | UInt32 | UInt40 | UInt48 | Int8 | Int16 | Int24 | Int32 | Int40 | Int48, consumed: number } | null | undefined {
```

Comment:
Hmm, why is undefined added to the return type?  It shouldn't be...

Reply:
Ah, it is because this function isn't actually finished.  :)  It was half way through authoring this function that I realized decoding was going to be a huge PITA and stopped to evaluate whether pursuing this entire branch was worthwhile or not.

Reply:
Master seems to work fine. You are awesome for trying to do the encode/decode but it is super likely the consumer of this lib will have the necessary deps at the ready.